### PR TITLE
[Languages]: Building a Screen Recording Application Using JavaScript

### LINKED ISSUE #5470: [Languages]: Building a Screen Recording Application Using JavaScript
*NOTE*: All **1st-time contributors** should know Topic approval and PR submission does NOT guarantee your Topic/Article will be published. Our team of Peer Reviewers and Content Moderators will review all PRs that come in to make sure they adhere to the standard of quality we expect from the EngEd community. 

We expect all community memebers to go through provided examples, resources, and previously published material before submitting content. As a rule of thumb, please only submit articles (pull requests) that are complete, formatted correctly, and include a fully polished article (ready to be published and error free). 

All revisions and edits should be *completed on your own forked repo (as to not take up room in the queue)*. **Any PR that is submitted incomplete (meaning not ready to be published as is) will be closed.** A PR being closed does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

### Topic Suggestion
Building a Screen Recording Application Using JavaScript

#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published (subject to vary depending on _volume and/or backlog of articles_).

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Writing sample(s): 
Include any links or writing samples - to help our team better gauge your writing quality.

#### Proposed title of article
Building a Screen Recording Application Using JavaScript

Use keyword research to improve your article's SEO.

#### Proposed article introduction
We use our computers every day for a variety of tasks. At times, we might need to record our actions on these devices. This is where the screen recorder applications become useful. The screen recorder applications are useful when we need to quickly capture our screen to create videos. It could be any meeting or a call. JavaScript being a popular language has the tools and capability needed to develop such apps in the browser.
In this tutorial, we will build an app that can record the user's screen and save the recording to the server.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
We will learn:

- How to build a system that records a video or audio
- Implementing the view and download to the user
- Compression on the server without losing quality

#### Article quality
This article is beginner-friendly using the popular MERN stack. We will learn useful concepts of JavaScript and server-side programming.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.
N/A

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Qodestackr**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5470

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Qodestackr** [comment]
This fixes #5470

**Comment by katungi** [comment]
Hello @Qodestackr , Glad to see you are back to give the community more content. Please remember to fill in the checkboxes in the PR as you wait for an available member of the Peer Review team to jump on the PR.

**Comment by mercymeave** [comment]
Hey @katungi, let me jump on this. I love javascript :smile:

**Comment by katungi** [comment]
Hey @mercymeave , sure thing. Have fun 🥳

**Review by mercymeave — COMMENTED** [review]
Great article @Qodestackr 
Proceeding to tech review.

**Review by mercymeave — APPROVED** [review]
Great article @Qodestackr  :rocket:

**Comment by WanjaMIKE** [comment]
Plagiarism check done.

File: content/articles/javascript-screen-recorder/index.md
Comment (mercymeave) [inline content/articles/javascript-screen-recorder/index.md:144]: This statement is slightly off. Correct it then we get done.

**Review by mercymeave — APPROVED** [review]
Great article @Qodestackr 

Let us wait for the final review.

File: content/articles/javascript-screen-recorder/index.md
Comment (LinusMuema) [inline content/articles/javascript-screen-recorder/index.md:109]: I think there is something missing here. Either an image or a code snippet. 
`we should see something like:`???

Please check on this as the article is ready to be published

**Review by LinusMuema — CHANGES_REQUESTED** [review]
Hi @Qodestackr, @mercymeave 
Please check on this issue

File: content/articles/javascript-screen-recorder/index.md
Comment (mercymeave) [reply content/articles/javascript-screen-recorder/index.md:109]: on it, sorry must have missed that.

**Review by LinusMuema — APPROVED** [review]

**Pull request merged**
